src/App.tsx

```diff
@@ -1,61 +1,218 @@
+import React, { useState } from 'react';
+
+export type User = (typeof usersFromServer)[0];
+
+export interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  createdAt: Date;
+}
+
+export interface UserTodos {
+  userId: number;
+  userName: string;
+  userEmail: string;
+  userLogin: string;
+  todoPosts: Todo[];
+}
+
+const usersTodos = usersFromServer
+  .map(user => {
+    const userTodo = todosFromServer.reduce((todosAcc: Todo[], todo) => {
+      if (user.id === todo.userId) {
+        todosAcc.push({ ...todo, ...{ createdAt: new Date() } });
+
+        return todosAcc;
+      }
+
+      return todosAcc;
+    }, []);
+
+    return {
+      userId: user.id,
+      userName: user.name,
+      userEmail: user.email,
+      userLogin: user.username,
+      todoPosts: userTodo,
+    };
+  })
+  .filter(userTodo => userTodo.todoPosts.length > 0);
+
+function findTheBiggestPostId(postList: UserTodos[]): number {
+  return Math.max(
+    ...postList.map(userPost =>
+      Math.max(...userPost.todoPosts.map(post => post.id)),
+    ),
+  );
+}
 
 export const App = () => {
+  // #region baseStates
+  const [currentUsersTodos, setCurrentUsersTodos] =
+    useState<UserTodos[]>(usersTodos);
+  const [postCount, setPostCount] = useState(
+    findTheBiggestPostId(currentUsersTodos) + 1,
+  );
+  // #endregion
+  // #region title
+  const [title, setTitle] = useState('');
+  // #endregion
+  // #region userSelecting
+  const [isUserSelected] = useState(false);
+  const [userSelectingValue, setUserSelectingValue] = useState<number>(0);
+  // #endregion
+
+  // #region fieldErrors
+  const [hasTitleAnError, setHasTitleAnError] = useState<boolean>(false);
+  const [hasSelectedUserAnError, setHasSelectedUserAnError] =
+    useState<boolean>(false);
+
+  const titleFieldHasAnError = title.length === 0;
+  const selectedUserFieldHasAnError = userSelectingValue === 0;
+  const formHasSomeError = hasTitleAnError || hasSelectedUserAnError;
+  // #endregion
+
+  // #region handlers
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleAnError(false);
+  };
+
+  const handleSelectingUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserSelectingValue(+event.target.value);
+    setHasSelectedUserAnError(false);
+  };
+
+  const handleSelectedUser = (userId: number): User => {
+    return usersFromServer.find(user => user.id === userId) as User;
+  };
+
+  const onSubmit = (selectedUser: User): void => {
+    const existingUser = currentUsersTodos.find(
+      user => user.userId === selectedUser.id,
+    );
+
+    const newTodo = {
+      id: postCount,
+      title: title,
+      completed: false,
+      userId: selectedUser.id,
+      createdAt: new Date(),
+    };
+
+    if (existingUser) {
+      existingUser.todoPosts.push(newTodo);
+
+      return setCurrentUsersTodos([...currentUsersTodos]);
+    }
```

Comment:
What is the point of doing it? And what is the reason of `existingUser` if you don't use this variable in the future

Reply:
if current user has any post in his `todoPost` list, then I just add one more post in his post's history insted of creating the same user with new post